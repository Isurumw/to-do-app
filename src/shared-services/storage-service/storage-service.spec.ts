import AsyncStorage from '@react-native-community/async-storage';
import StorageService from './storage-service';

const storage = new StorageService(AsyncStorage);

describe('check the behaviour of the storage service wrapper', () => {
  it('check method GET for a supplied key', async () => {
    AsyncStorage.getItem = jest.fn(() => Promise.resolve('test_value'));

    await storage.get('test_key');
    expect(AsyncStorage.getItem).toHaveBeenCalledTimes(1);
    expect(AsyncStorage.getItem).toHaveBeenCalledWith('test_key');
  });

  it('check method GET, when there is an error', async () => {
    AsyncStorage.getItem = jest.fn(() =>
      Promise.reject(new Error('Failed to fetch the value')),
    );

    expect(storage.get('test_key')).rejects.toMatchSnapshot();
  });

  it('check method Set, when trying to store an item in the storage', async () => {
    const mockUser = {
      firstName: 'first_name',
      lastName: 'last_name',
    };
    AsyncStorage.setItem = jest.fn();

    await storage.set('user', JSON.stringify(mockUser));

    expect(AsyncStorage.setItem).toHaveBeenCalledTimes(1);
    expect(AsyncStorage.setItem).toHaveBeenCalledWith(
      'user',
      JSON.stringify(mockUser),
    );
  });
});
