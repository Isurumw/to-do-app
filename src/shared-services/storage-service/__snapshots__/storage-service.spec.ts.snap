// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the behaviour of the storage service wrapper check method GET, when there is an error 1`] = `[Error: Could not receive the test_key from cache]`;
